go-controller/pkg/ovn/libovsdbops/nat.go

```diff
@@ -0,0 +1,123 @@
+package libovsdbops
+
+import (
+	"fmt"
+	"net"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+	"github.com/ovn-org/libovsdb/model"
+	libovsdb "github.com/ovn-org/libovsdb/ovsdb"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+)
+
+func buildRouterNAT(
+	natType nbdb.NATType,
+	externalIP net.IP,
+	logicalIP net.IP,
+	logicalPort string,
+	externalMac string,
+	) *nbdb.NAT {
+	return &nbdb.NAT{
+		Type:            natType,
+		ExternalIP:      externalIP.String(),
+		LogicalIP:       logicalIP.String(),
+		LogicalPort:     []string{logicalPort},
+		ExternalMAC:     []string{externalMac},
+		Options:         map[string]string{"stateless": "false"},
+	}
+}
+
+// TODO(flaviof): move to router.go
+func findRouter(nbClient libovsdbclient.Client, routerName string) (string, error) {
+	routers := []nbdb.LogicalRouter{}
+	err := nbClient.WhereCache(func(item *nbdb.LogicalRouter) bool {
+		return item.Name == routerName
+	}).List(&routers)
+
+	if err != nil {
+		return "", fmt.Errorf("error finding logical router %s: %v", routerName, err)
+	}
+
+	if len(routers) == 0 {
+		return "", fmt.Errorf("found no logical routers with name %s", routerName)
+	}
+
+	if len(routers) > 1 {
+		return "", fmt.Errorf("unexpectedly found multiple logical routers: %+v", routers)
+	}
+
+	return routers[0].UUID, nil
+}
+
+// TODO: findNat   return  []nbdb.NAT{} ?
+
+// TODO: addNatsToRouterOps  input []libovsdb.Operation ops, []string natUuids   return ops
```

Comment:
@jcaamano pls help me with how these 2 TODO functions should look like^^

Reply:
References: 

https://github.com/jcaamano/ovn-kubernetes/blob/3f7c4fd9e8b8d46b784c897b16041b67ae5545da/go-controller/pkg/ovn/libovsdbops/acl.go#L33
https://github.com/jcaamano/ovn-kubernetes/blob/3f7c4fd9e8b8d46b784c897b16041b67ae5545da/go-controller/pkg/ovn/libovsdbops/portgroup.go#L13
https://github.com/jcaamano/ovn-kubernetes/blob/3f7c4fd9e8b8d46b784c897b16041b67ae5545da/go-controller/pkg/ovn/libovsdbops/portgroup.go#L204